Revise hardware recommendations (#896)
rancher/rancher-docs: docs/pages-for-subheaders/installation-requirements.md

If you plan to run Rancher on ARM64, see [Running on ARM64 (Experimental).](../how-to-guides/advanced-user-guides/enable-experimental-features/rancher-on-arm64.md)

### RKE2 Specific Requirements

> For the container runtime, RKE2 bundles its own containerd. Docker is not required for RKE2 installs.

Reviewer: ```suggestion
RKE2 bundles its own container runtime, containerd. Docker is not required for RKE2 installs.
```

If you are installing Rancher on a K3s cluster with Alpine Linux, follow [these steps](https://rancher.com/docs/k3s/latest/en/advanced/#additional-preparation-for-alpine-linux-setup) for additional setup.

### RKE Specific Requirements

For the container runtime, RKE should work with any modern Docker version.

For more information see [Installing Docker,](../getting-started/installation-and-upgrade/installation-requirements/install-docker.md)

## Hardware Requirements

The following sections describe the CPU, memory, and I/O requirements for nodes where Rancher is installed based on the size of the infrastructure it manages.

> ### Premise

Author: Addressed with d37a50c041e

 - size of managed infrastructure (eg. node count, cluster count)
 - complexity of the desired access control rules (eg. `RoleBinding` object count)
 - number of workloads (eg. Kubernetes deployments, Fleet deployments)
 - usage patterns (eg. subset of functionality actively used, frequency of use, number of concurrently active users)

Because of the high number of influencing factors and their variability over time, requirements in this document have to be interpreted as reasonable starting points that worked acceptably well in most observed use cases so far. Nevertheless, particular use cases may differ - with higher or lower requirements. For enquiries about a specific scenario please [contact Rancher](https://rancher.com/contact/) for further guidance.

In particular, requirements on this page are subject to "typical use" conditions, which include:
 - total count of Kubernetes resources under 60 thousand per type
 - up to 120 pods per node
 - up to 200 CRDs in the upstream (local) cluster
 - up to 100 CRDs in downstream clusters
 - up to 50 Fleet deployments
Higher numbers are possible but requirements might be higher.

> Note that visualization in the UI might be impacted for users with visibility on all resources in a type above 20 thousand total items.

Author: Updated as 7bb0daeb060, please comment again if wording does not sound right. Thanks

> :::note Evolution:

Author: ![image](https://github.com/rancher/rancher-docs/assets/250541/9be23fff-a66d-4669-8a44-4c4ae3eab659)

Rendering seems correct here, I am marking this conversation resolved.

Rancher's code base evolves, use cases change, and the body of accumulated Rancher experience grows every day.

Because of that, this document's recommendations are subject to change over time with the aim of providing increasingly more concrete and accurate guidelines.

If you find your Rancher deployment does not comply with recommendations in this document, while it used to comply with earlier revisions, you can [contact Rancher](https://rancher.com/contact/) for an ad-hoc evaluation.

:::

### RKE2 Kubernetes

Minimum CPU and memory requirements for each individual node in the [Kubernetes cluster Rancher is installed in](install-upgrade-on-a-kubernetes-cluster.md) are listed in the table below.

> Please note that a highly available setup with at least 3 nodes is required for all production usages.

Reviewer: ```suggestion
Please note that a highly available setup with at least three nodes is required for production.
```

> | Managed Infrastructure Size | Clusters  | Nodes      | vCPUs | RAM   |

Reviewer: ```suggestion
| Managed Infrastructure Size | Clusters  | Nodes      | vCPUs | RAM   |
```

Instead of "Up to" in each cell, revise to have the header row state "Maximum # of Clusters" & "Maximum # of Nodes"

Author: Addressed as of 548e4de67ae, resolving.

|-----------------------------|-----------|------------|-------|-------|
| Small                       | Up to 150 | Up to 1500 | 4     | 16 GB |
| Medium                      | Up to 300 | Up to 3000 | 8     | 32 GB |
| Large (*)                   | Up to 500 | Up to 5000 | 16    | 64 GB |
| Larger (†)                  | (†)       | (†)        | (†)   | (†)   |

> (*): Large deployments require [additional tuning and following of best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md).

Reviewer: ```suggestion
(*): Large deployments should [follow best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md) for running Rancher at scale.
```

Author: I would like to use an assertive tone here - we already know many or most large/larger cases will just not work unless best practices are followed (and there is an ongoing discussion about supportability rules in the context of Rancher Prime subscribers).

Without going into that discussion here, thinking about open source community members, how can we phrase the concept that best practices are generally, well, best practices - but they turn into something that is more of a requirement in order to have a chance of good performance at high scale?

Author: How about
```suggestion
(*): Large deployments are also expected to [follow best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md) for adequate performance.
```

Reviewer: "should also" > "require that you"

> (†): Depending on various factors, larger deployment sizes are generally possible with ad-hoc hardware recommendations and tuning. You can [contact Rancher](https://rancher.com/contact/) for a customized evaluation.

Reviewer: ```suggestion
(†): Larger deployment sizes are generally possible with ad-hoc hardware recommendations and tuning. You can [contact Rancher](https://rancher.com/contact/) for a custom evaluation.
```

Refer to RKE2 documentation for more detailed information on [RKE2 general requirements](https://docs.rke2.io/install/requirements).

### K3s Kubernetes

> Minimum CPU and memory requirements for each individual node in the [Kubernetes cluster Rancher is installed in](install-upgrade-on-a-kubernetes-cluster.md) are listed in the table below.

Reviewer: ```suggestion
The following table lists minimum CPU and memory requirements for each node in the [upstream cluster](install-upgrade-on-a-kubernetes-cluster.md).
```

Please note that a highly available setup with at least 3 nodes is required for all production usages.

> | Managed Infrastructure Size | Clusters  | Nodes      | vCPUs | RAM   | External Database Host (†) |

Author: Addressed as of 548e4de67ae, resolving.

|-----------------------------|-----------|------------|-------|-------|----------------------------|
| Small                       | Up to 150 | Up to 1500 | 4     | 16 GB | 2 vCPUs, 8 GB + 1000 IOPS  |
| Medium                      | Up to 300 | Up to 3000 | 8     | 32 GB | 4 vCPUs, 16 GB + 2000 IOPS |
| Large (*)                   | Up to 500 | Up to 5000 | 16    | 64 GB | 8 vCPUs, 32 GB + 4000 IOPS |

> (*): Large deployments require [additional tuning and following of best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md).

Author: Pending disccussion in https://github.com/rancher/rancher-docs/pull/896/files#r1342999088

> (*): Large deployments require [additional tuning and following of best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md).

Author: Addressed as of dddf2b28c45

> (†): External Database Host refers to the optional possibility of hosting [k3s cluster data store on an external dedicated host](https://docs.k3s.io/datastore). Exact requirements will depend on the chosen data store, this table is a guideline only.

Reviewer: ```suggestion
(†): External Database Host refers to hosting the K3s cluster data store on an [dedicated external host](https://docs.k3s.io/datastore). This is optional. Exact requirements depend on the external data store.
```

Refer to k3s documentation for more detailed information on [k3s general requirements](https://docs.k3s.io/installation/requirements).

### Hosted Kubernetes

Minimum CPU and memory requirements for each individual node in the [Kubernetes cluster Rancher is installed in](install-upgrade-on-a-kubernetes-cluster.md) are listed in the table below.

> Please note that a highly available setup with at least 3 nodes is required for all production usages.

Reviewer: ```suggestion
Please note that a highly available setup with at least three nodes is required for production.
```

> These requirements apply hosted Kubernetes clusters such as EKS, AKS, or GKE. They do not apply to Rancher SaaS solutions such as Rancher Prime Hosted](https://www.rancher.com/products/rancher).

Reviewer: ```suggestion
These requirements apply to hosted Kubernetes clusters such as Amazon Elastic Kubernetes Service (EKS), Azure Kubernetes Service  (AKS), or Google Kubernetes Engine (GKE). They don't apply to Rancher SaaS solutions such as [Rancher Prime Hosted](https://www.rancher.com/products/rancher).
```

> | Managed Infrastructure Size | Clusters  | Nodes      | vCPUs | RAM   |

Author: Addressed as of 548e4de67ae, resolving.

|-----------------------------|-----------|------------|-------|-------|
| Small                       | Up to 150 | Up to 1500 | 4     | 16 GB |
| Medium                      | Up to 300 | Up to 3000 | 8     | 32 GB |
| Large (*)                   | Up to 500 | Up to 5000 | 16    | 64 GB |

> (*): Large deployments require [additional tuning and following of best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md).

Author: https://github.com/rancher/rancher-docs/pull/896/files#r1342999088

> (*): Large deployments require [additional tuning and following of best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md).

Author: Addressed as of dddf2b28c45

### RKE

> Minimum CPU and memory requirements for each individual node in the [Kubernetes cluster Rancher is installed in](install-upgrade-on-a-kubernetes-cluster.md) are listed in the table below.

Reviewer: ```suggestion
The following table lists minimum CPU and memory requirements for each node in the [upstream cluster](install-upgrade-on-a-kubernetes-cluster.md).
```

> Please note that a highly available setup with at least 3 nodes is required for all production usages.

Reviewer: ```suggestion
Please note that a highly available setup with at least three nodes is required for production.
```

> | Managed Infrastructure Size | Clusters  | Nodes      | vCPUs | RAM   |

Reviewer: Instead of "Up to" in each cell, revise to have the header row state "Maximum # of Clusters" & "Maximum # of Nodes"

Author: Addressed as of 548e4de67ae, resolving.

|-----------------------------|-----------|------------|-------|-------|
| Small                       | Up to 150 | Up to 1500 | 4     | 16 GB |
| Medium                      | Up to 300 | Up to 3000 | 8     | 32 GB |
| Large (*)                   | Up to 500 | Up to 5000 | 16    | 64 GB |

> (*): Large deployments require [additional tuning and following of best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md).

Reviewer: ```suggestion
(*): Large deployments should [follow best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md) for running Rancher at scale.
```

Author: https://github.com/rancher/rancher-docs/pull/896/files#r1342999088

Author: Addressed as of dddf2b28c45148a1246891066e418577238e725d

> Refer to RKE documentation for more detailed information on [RKE general requirements](https://rke.docs.rancher.com/os).

Reviewer: ```suggestion
Refer to the RKE documentation for more detailed information on [general requirements](https://rke.docs.rancher.com/os).
```

### Docker

> Minimum CPU and memory requirements for a [single node installation of Rancher](rancher-on-a-single-node-with-docker.md) are listed in the table below.

Reviewer: ```suggestion
The following table lists minimum CPU and memory requirements for a [single Docker node installation of Rancher](rancher-on-a-single-node-with-docker.md).
```

Please note that a Docker installation is only suitable for development or testing purposes and is not meant to be used in production environments.

> | Managed Infrastructure Size | Clusters | Nodes     | vCPUs | RAM  |

Reviewer: Instead of "Up to" in each cell, revise to have the header row state "Maximum # of Clusters" & "Maximum # of Nodes"

Author: Addressed as of 548e4de67ae, resolving.

If you are installing Rancher on a K3s cluster with Alpine Linux, follow [these steps](https://rancher.com/docs/k3s/latest/en/advanced/#additional-preparation-for-alpine-linux-setup) for additional setup.

### RKE Specific Requirements

> For the container runtime, RKE should work with any modern Docker version.

Reviewer 2: We have validated Docker versions for RKE provisioned clusters in the support matrix. Should this be called out instead of any modern? https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-6/

Author: Addressed as of 421e12c05e9

## Hardware Requirements

The following sections describe the CPU, memory, and I/O requirements for nodes where Rancher is installed. Requirements vary based on the size of the infrastructure.

### Practical Considerations

Rancher's hardware footprint depends on a number of factors, including:

 - Size of the managed infrastructure (e.g., node count, cluster count).
 - Complexity of the desired access control rules (e.g., `RoleBinding` object count).
 - Number of workloads (e.g., Kubernetes deployments, Fleet deployments).
 - Usage patterns (e.g., subset of functionality actively used, frequency of use, number of concurrent users).

Since there are a high number of influencing factors that may vary over time, the requirements listed here should be understood as reasonable starting points that work well for most use cases. Nevertheless, your use case may have different requirements. For inquiries about a specific scenario please [contact Rancher](https://rancher.com/contact/) for further guidance.

In particular, requirements on this page are subject to typical use assumptions, which include:

>  - Under 60 thousand total Kubernetes resources, per type.

Reviewer 2: minor note - maybe change to 60,000 for consistency

Author: Addressed as of d5b649a6ca9

 - Up to 120 pods per node.
 - Up to 200 CRDs in the upstream (local) cluster.
 - Up to 100 CRDs in downstream clusters.
 - Up to 50 Fleet deployments.

> Higher numbers are possible but requirements might be higher. If you have more than 20 thousand resources of the same type, loading time of the whole list through the Rancher UI might take several seconds.

Reviewer 2: same here- 20,000

Author: Addressed as of d5b649a6ca9

| Small                       | 150                        | 1500                    | 4     | 16 GB |
| Medium                      | 300                        | 3000                    | 8     | 32 GB |
| Large (*)                   | 500                        | 5000                    | 16    | 64 GB |
| Larger (†)                  | (†)                        | (†)                     | (†)   | (†)   |

(*): Large deployments require that you [follow best practices](../reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md) for adequate performance.

(†): Larger deployment sizes are generally possible with ad-hoc hardware recommendations and tuning. You can [contact Rancher](https://rancher.com/contact/) for a custom evaluation.

Refer to RKE2 documentation for more detailed information on [RKE2 general requirements](https://docs.rke2.io/install/requirements).

### K3s Kubernetes

The following table lists minimum CPU and memory requirements for each node in the [upstream cluster](install-upgrade-on-a-kubernetes-cluster.md).

Please note that a highly available setup with at least three nodes is required for production.

> | Managed Infrastructure Size | Maximum number of Clusters | Maximum number of Nodes | vCPUs | RAM   | External Database Host (†) |

Author: Right, addressed in 79e276d8cc2, thanks
